terraform/modules/data-share-service/lb.tf

```diff
@@ -15,12 +21,45 @@ resource "random_id" "http_sufix" {
   byte_length = 2
 }
 
-#TODO-https://github.com/alphagov/gdx-data-share-poc/issues/20: When we have our own route53 we can lock this to HTTPS
+resource "aws_acm_certificate" "acm_lb_certificate" {
+  domain_name       = var.hosted_zone_name
+  validation_method = "DNS"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_route53_record" "lb_cert_validation" {
+  for_each = {
+    for dvo in aws_acm_certificate.acm_lb_certificate.domain_validation_options : dvo.domain_name => {
+      name   = dvo.resource_record_name
+      record = dvo.resource_record_value
+      type   = dvo.resource_record_type
+    }
+  }
+
+  allow_overwrite = true
+  name            = each.value.name
+  records         = [each.value.record]
+  ttl             = 60
+  type            = each.value.type
+  zone_id         = var.hosted_zone_id
+}
+
+resource "aws_acm_certificate_validation" "acm_lb_certificate_validation" {
+  certificate_arn         = aws_acm_certificate.acm_lb_certificate.arn
+  validation_record_fqdns = [for record in aws_route53_record.lb_cert_validation : record.fqdn]
+}
+
 #tfsec:ignore:aws-elb-http-not-used
-resource "aws_lb_listener" "listener_http" {
+resource "aws_lb_listener" "listener_https" {
```

Comment:
could you add a moved resource to say that listener_http was moved to listener_https pls (and for test_listener_http(s))